### PR TITLE
Adding hold and unhold feature to media

### PR DESCRIPTION
**Pull request by v-vdharmaraj**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

**Testing:**
1. Hold the target participant with or without the playsource for few seconds 
2. checking the is_on_hold property from the get participant => expected to be true and actual also true
3. Unhold the target participant 
4. checking the is_on_hold property from the get participant => expected to be false and actual also false

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-callautomation/tests/test_call_media_client.py [hunk 1]
```diff
@@ -413,3 +415,81 @@ def test_update_transcription(self):
...
+                                          operation_context=operation_context)
+
+        expected_hold_request = HoldRequest(
+            target_participant=[serialize_identifier(self.target_user)],
+            operation_context=operation_context
+        )
+        mock_hold.assert_called_once()
+        actual_hold_request = mock_hold.call_args[0][1]
+
+        self.assertEqual(expected_hold_request.operation_context, actual_hold_request.operation_context)
```
Comment (cochi2) [inline sdk/communication/azure-communication-callautomation/tests/test_call_media_client.py:478]: Please, explicitly verify that the optional fields are missing as expected.

**Comment by cochi2** [comment]
This PR is missing the `async` version of `Hold` and `Unhold` methods.

**Comment by cochi2** [comment]
This PR is missing changes for the [CallParticipant](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/communication/azure-communication-callautomation/azure/communication/callautomation/_models.py#L531) object to include `is_on_hold`.

File: sdk/communication/cspell.json [hunk 2]
```diff
@@ -17,6 +17,8 @@
         "Nify",
         "deserializers",
         "ctxt",
-        "wday"
+        "wday",
+		"unhold",
+		"Unhold"
```
Comment (Paresh-Arvind-Patil) [inline sdk/communication/cspell.json:22]: Nit: fix indentation here

**Review by Paresh-Arvind-Patil — APPROVED** [review]

**Pull request merged**